params/config.go

```diff
@@ -509,10 +518,22 @@ func (c *ChainConfig) AvalancheRules(blockNum, blockTimestamp *big.Int) Rules {
 	rules := c.rules(blockNum)
 
 	rules.IsSubnetEVM = c.IsSubnetEVM(blockTimestamp)
-	rules.IsAllowListEnabled = c.IsAllowList(blockTimestamp)
+
+	precompileMap := make(map[string]bool, len(c.enabledStatefulPrecompiles()))
+	for _, enabledPrecompile := range c.enabledStatefulPrecompiles() {
+		precompileMap[enabledPrecompile.Name()] = c.IsPrecompileActivated(enabledPrecompile, blockTimestamp)
+	}
```

Comment:
`enabledStatefulPrecompiles()` should only return non-nil configs that need to be enabled. This is so that we can do as few checks as possible for them when processing blocks. We can also order them in the future and cache it, so that we don't do this each time and can stop as soon as we hit a timestamp that doesn't need to be activated instead of iterating through all of them every time.

Reply:
changed bool to struct. `enabledStatefulPrecompiles()` already returns non-nil configs. My intention was to reduce the need LOC to add a precompile. With this we can tell developers to add their precompile to `enabledStatefulPrecompiles` with proper code and be done with it. Otherwise they need to add precompile to `Rule` struct + `enabledStatefulPrecompiles()` + `AvalancheRules` + another function for `IsEnabled`.